lib/SIL/IR/SILSymbolVisitor.cpp

```diff
@@ -608,6 +598,21 @@ class SILSymbolVisitorImpl : public ASTVisitor<SILSymbolVisitorImpl> {
       }
     }
 
+    // Statically/globally stored variables have some special handling.
+    // Global accessor gets no linkage limit in the same resilience domain,
+    // e.g. in package with non-resilient access optimizations enabled,
+    // so it should be added in such case.
+    if (!VD->isResilient() ||
+        VD->getModuleContext()->allowNonResilientAccess()) {
```

Comment:
To avoid diverging in the future can you just remove the outermost `if` and unconditionally visit symbols associated with a global? If SILDeclRef says they're not public we ignore them so there's no harm in doing so.

Reply:
done